be/src/service/service_be/lake_service.cpp

```diff
@@ -289,6 +290,11 @@ void LakeServiceImpl::publish_version(::google::protobuf::RpcController* control
     TEST_SYNC_POINT("LakeServiceImpl::publish_version:return");
 }
 
+struct PublishRequestCtx {
```

Comment:
`PublishRequestCtx` and the following `PendingCompactRequestCtx` are both for resource management,  it can be designed to be used as generic. One is to use template

```
template<typename PBType>
struct RequestCtx {
    std::unique_ptr<brpc::Controller> cntl;
    std::unique_ptr<PBType> resp;
};
```

or using base message type

```
struct RequestCtx {
    std::unique_ptr<brpc::Controller> cntl;
    std::unique_ptr<protobuf::Message> resp;
};
```